components/brave_extension/extension/brave_extension/background/api/browserActionAPI.ts

```diff
@@ -48,12 +49,16 @@ export function init () {
  * @param {string} text - The text to put on the badge
  */
 export const setBadgeText = (tabId: number, text: string) => {
-  if (chrome.browserAction) {
-    chrome.browserAction.setBadgeText({
-      tabId,
-      text: String(text)
+  getViewPreferences()
```

Comment:
Calling in to another API from this API doesn't seem like great code organization to me for the following reasons:
1. It's retrieving the data again, adding a (small) delay.
2. The app decision logic is now in multiple places.
3. It's complicating the otherwise simple module, perhaps also complicating testability. This means the API no longer just does what it says `setBadgeText`.

Instead, since we store this data in the redux state or some other main decision making area, can we simply read current value of the state before we call `setBadgeText` and decide what to pass in?

Reply:
that one is tricky because change happens with shields closed so there's no `state` since the component isn't mounted. I moved the data to the persistent state (shields localStorage). lmk if you have a better option